vignettes/tidyselect.Rmd

```diff
@@ -0,0 +1,310 @@
+---
+title: "Implementing tidyselect interfaces"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Implementing tidyselect interfaces}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup}
+library(tidyselect)
+library(magrittr)
+```
+
+```{r, include = FALSE}
+# For better printing
+mtcars <- tibble::as_tibble(mtcars)
+iris <- tibble::as_tibble(iris)
+
+options(
+  tibble.print_min = 4,
+  tibble.print_max = 4
+)
+```
+
+tidyselect implements a specialised sublanguage of R for selecting
+variables from data frames and other data structures. A technical
+description of the DSL is available in the [syntax
+vignette](https://tidyselect.r-lib.org/articles/syntax.html). In this
+vignette, we describe how to include tidyselect in your own packages.
+
+
+## Before we start
+
+### Selections as dots or as named arguments
+
+There are two major ways of designing a function that takes
+selections.
+
+- __Variadic dots__ as in `dplyr::select()`.
+
+    ```{r, eval = FALSE}
+    mtcars %>% dplyr::select(mpg, cyl)
+    ```
+
+- __Named arguments__ as in `tidyr::pivot_longer()`. In this case,
+  multiple inputs can be provided inside `c()` or by using boolean
+  operators:
+
+    ```{r, eval = FALSE}
+    mtcars %>% pivot_longer(c(mpg, cyl))
+    mtcars %>% pivot_longer(mpg | cyl)
+    ```
+
+Our general recommendation is to take dots when the main purpose of
+the function is to create a new data structure based on a selection.
+When the selection is accessory to the main purpose of the function,
+take it as a named argument. For more advice about this, see the
```

Comment:
If in doubt, we recommend using named arguments, because it's easier to go from named arg -> dots than dots -> named arg.